server/src/main/java/org/elasticsearch/common/settings/AddStringKeyStoreCommand.java

```diff
@@ -83,8 +85,19 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
 
         final char[] value;
         if (options.has(stdinOption)) {
-            BufferedReader stdinReader = new BufferedReader(new InputStreamReader(getStdin(), StandardCharsets.UTF_8));
-            value = stdinReader.readLine().toCharArray();
+            try (BufferedReader stdinReader = new BufferedReader(new InputStreamReader(getStdin(), StandardCharsets.UTF_8));
+                 CharArrayWriter writer = new CharArrayWriter()) {
+                int charInt;
+                while ((charInt = stdinReader.read()) != -1) {
+                    if ((char) charInt == '\r' || (char) charInt == '\n') {
+                        break;
+                    }
+                    writer.write((char) charInt);
+                }
+                value = writer.toCharArray();
+            } catch (IOException e) {
+                throw new UserException(ExitCodes.DATA_ERROR, "Error reading the value from stdin", e);
```

Comment:
This message is still opaque to a user. Remember UserException does *not* print a stacktrace. In this case, I think we should just let IOException bubble up and print the full stacktrace? What benefit is there in wrapping it? If there is a subclass of io errors that the user can directly do something about, then we should distinguish that and message appropriately with actions the user can take.

Reply:
I got the idea to add the exceptions to the UserException from Jay's feedback in https://github.com/elastic/elasticsearch/pull/38498#discussion_r260391108 but I see what you're saying here. There is no subset of i/o errors that I know of where we could print something useful for the user, so I'll let the IOException be thrown